### ISSUE TITLE
get_status_display only returns the status not the LOAN_STATUS value

### ISSUE THREAD

**Issue by abdulwahidgul24085** [mdn/django-locallibrary-tutorial#61]
In the 5th tutorial when I was working on in the template section when we are looping over the book instances. I notice that the status and the display_status both had the same return value as "a" instead of availability. I checked the code and the display_status does only return self.status. I assumed that we would need a complete word from the LOAN_STATUS. So I modified the code as follows

```
def get_status_display(self):
        for key, value in enumerate(self.LOAN_STATUS):
            if value[0] == self.status:
                return self.LOAN_STATUS[key][1]
        return None
```

I hope this helps if anyone is interested in that.

**Comment by hamishwillee** [comment]
Thanks very much. Can you please link to the tutorial page and the page in this repo that is affected. That will make this easier to review and update if needed.

**Comment by abdulwahidgul24085** [comment]
Sure. So in tutorial # 6 [(Creating Generic List and Detail Views)](https://developer.mozilla.org/en-US/docs/Learn/Server-side/Django/Generic_views#What_does_it_look_like_2) in this output you can see the output shows the complete value for the LOAN_STATUS. 

You can need to add the `get_status_display` function in your code since I see it talked about in the tutorial and does not exist in the `BookInstance` class in [catalog/models.py](https://github.com/mdn/django-locallibrary-tutorial/blob/acf2a446cdb45edd9105303b15dd86fca4458c85/catalog/models.py#L66)
I hope this clarify's the confusion, and points to all the resources that need to be amended.

P.S. I hope you merge my code. It will make it my first official open-source contributed code. A man can dream can't he lol. Also, great tutorial. One of the most comprehensive introductions to Django that I could find.

**Comment by dlyrium** [comment]
get_FOO_display() is a method in the django.db.Models class. It should work fine, without any extra code.

**Comment by hamishwillee** [comment]
Hi @abdulwahidgul24085 ,
Thanks for this, and sorry for the delay getting back to you.

As @dlyrium say, this is not needed. The `BookInstance` model has a [choices](https://docs.djangoproject.com/en/3.1/ref/models/fields/) field named `status`. This includes a definition of the choices with a key and a human readable description.
The key is what is stored, and wherever a choices is used, Django automatically adds a function to get the human readable text. This is named `get_FOO_display()` where FOO is the name of the field (so in this case `get_status_display()`

This is not obvious at all, so I have updated the [documentation on generic views](https://wiki.developer.mozilla.org/en-US/docs/Learn/Server-side/Django/Generic_views)

Closing now. Thanks - reporting the confusion is a useful contribution!
